third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -130,78 +224,90 @@ void dds_device::impl::open( const dds_stream_profiles & profiles )
-        { "value", new_value },
-    };
-
-    write_control_message( j );
-
-    std::unique_lock< std::mutex > lock( _option_mutex );
-    if( ! _option_cv.wait_for( lock, std::chrono::milliseconds( _message_timeout_ms ), [&](){
-        return !_option_response_queue.empty() && _option_response_queue.front()["counter"] == this_message_counter;
-    } ) )
-        throw std::runtime_error( "Did not receive reply while setting option " + option->get_name() );
-
-    auto response = _option_response_queue.front();
-    _option_response_queue.pop();
-    if( rsutils::json::get< bool >( response, "successful" ) )
-        option->set_value( new_value );
-    else
-        throw std::runtime_error( rsutils::json::get< std::string >( response, "failure-reason" ) );
+    nlohmann::json j = nlohmann::json::object({
+        { id_key, id_set_option },
+        { option_name_key, option->get_name() },
+        { value_key, new_value }
+    });
+    if( ! option->owner_name().empty() )
+        j[owner_name_key] = option->owner_name();
+
+    nlohmann::json reply;
+    write_control_message( j, &reply );
+
+    if( rsutils::json::get( reply, status_key, status_ok ) != status_ok )
+        throw std::runtime_error(
+            rsutils::json::get< std::string >( reply, explanation_key, "failed to set-option; no explanation" ) );
+    //option->set_value( new_value );
 }
 
+
 float dds_device::impl::query_option_value( const std::shared_ptr< dds_option > & option )
 {
     if( !option )
         DDS_THROW( runtime_error, "must provide an option to query" );
 
-    uint32_t this_message_counter = _control_message_counter++;
-
-    nlohmann::json j = {
-        { "id", "query-option" },
-        { "counter", this_message_counter },
-        { "option-name", option->get_name() },
-        { "owner-name", option->owner_name() },
-    };
-
-    write_control_message( j );
+    nlohmann::json j = nlohmann::json::object({
+        { id_key, id_query_option },
+        { option_name_key, option->get_name() }
+    });
+    if( ! option->owner_name().empty() )
+        j[owner_name_key] = option->owner_name();
 
-    std::unique_lock< std::mutex > lock( _option_mutex );
-    if( ! _option_cv.wait_for( lock, std::chrono::milliseconds( _message_timeout_ms ), [&]() {
-        return !_option_response_queue.empty() && _option_response_queue.front()["counter"] == this_message_counter;
-    } ) )
-        throw std::runtime_error( "Did not receive reply while querying option " + option->get_name() );
+    nlohmann::json reply;
+    write_control_message( j, &reply );
 
-    auto response = _option_response_queue.front();
-    _option_response_queue.pop();
-    if( rsutils::json::get< bool >( response, "successful" ) )
-        return rsutils::json::get< float >( response, "value" );
-    else
-        throw std::runtime_error( rsutils::json::get< std::string >( response, "failure-reason" ) );
+    if( rsutils::json::get( reply, status_key, status_ok ) != status_ok )
```

Comment:
I am not sure I get this - if there is an error `get` will return the default value and the expression will evaluate to false and won't throw.

Reply:
Get the `status` value, and default to `ok`. If the result isn't `ok`, this is an error.